dadi/frontend/components/DocumentFilters/DocumentFilters.jsx

```diff
@@ -19,52 +19,75 @@ export default class DocumentFilters extends Component {
     } = this.props
 
     // Evaluate passed filter / store in state
+    let filters = (isValidJSON(filter) ? objectToArray(JSON.parse(filter), 'field') : []).map(this.deconstructFilters)
     this.state = {
-      edit: false,
       visible: filter || false,
-      filters: isValidJSON(filter) ? JSON.parse(filter) : null
+      filters: filters
     }
   }
   render() {
-    const {visible, filters, edit} = this.state
+    const {visible, filters} = this.state
     const {collection} = this.props
+
     return (
       <div>
         <Button onClick={this.toggleFilters.bind(this)}>Filters</Button>
-
-        {visible && filters && Object.keys(filters).length && collection && (
+        {visible && filters && collection && (
           <div>
-            {Object.keys(filters).map(key => ( 
+            {filters.map((filter, index) => ( 
               <DocumentFilter 
-                field={key} 
-                value={filters[key]} 
+                index={index}
+                field={filter.field} 
+                type={filter.type} 
+                value={filter.value} 
                 fields={collection.fields} 
                 updateFilter={this.updateUrlFilters.bind(this)} 
                 />
             ))}
           </div>
         )}
-        {edit && (
-          <DocumentFilter 
-            fields={collection.fields} 
-            updateFilter={this.updateUrlFilters.bind(this)} 
-          />
-        )}
         {visible && (
           <Button onClick={this.addFilter.bind(this)}>Add</Button>
         )}
       </div>
     )
   }
 
-  updateUrlFilters(filter) {
+  deconstructFilters(filter) {
+    // If there is no filter type and the field value is a string, add a type
+    if (typeof filter.value === 'string') return Object.assign(filter, {type: '$eq'})
+    else {
```

Comment:
This looks a bit odd. We usually put the body of the `if` on a new line if there is an `else` block, like:

```js
  deconstructFilters(filter) {
    // If there is no filter type and the field value is a string, add a type
    if (typeof filter.value === 'string') {
      return Object.assign(filter, {type: '$eq'})
    } else {
      let type = Object.keys(filter.value)[0]

      return Object.assign(filter, {type: type, value: filter.value[type]})
    }
  }
```